lib/cachex/actions/warm.ex

```diff
@@ -26,15 +26,22 @@ defmodule Cachex.Actions.Warm do
   be returned in the result of this call.
   """
   def execute(cache() = cache, options) do
+    respect_async_warmers = Keyword.get(options, :respect_async_warmers, false)
```

Comment:
I've been looking at this a lot and it makes sense, but I think I would prefer to avoid this option at all. I'm not totally sure if this will work, but I'm thinking:

1. All of the `async` handling comes out of warmers. Instead a warmer offers a `handle_call` and a `handle_info`, with the `handle_call` simply calling `handle_info` and returning the result.
2. The only place that `async` is considered is cache startup, so I'd rather have some private function in `cachex.ex` which walks the list and determines which to call based on the `async` flag.
3. I'm trying to decide if `Cachex.warm/2` should support `async` vs `sync`. Not sure but we can always add that later anyway. There's probably a pretty good argument that it should default to `sync` in the same way the initial warming does. 

It should be noted that `Cachex.warm/2` isn't actually public yet (unreleased), so we can change that however we need to. Maybe this is all out of scope for this PR and I can merge as-is, and follow up with the changes I'm talking about?

Reply:
I refactored this based on your suggestions - now this does not rely on `Cachex.warn` but has a private function that executes the warmers based on their async attribute